ingestion-edge/ingestion_edge/publish.py

```diff
@@ -131,34 +134,45 @@ def submit(q: SQLiteAckQueue, topic: str, **kwargs) -> Tuple[str, int]:
         if value is not None
     }
     try:
-        _publish(topic, data, attrs)
+        await _publish(topic, data, attrs)
     except TRANSIENT_ERRORS:
         # transient api call failure, write to queue
-        q.put((topic, data, attrs))
-    return "", 200
+        try:
+            q.put((topic, data, attrs))
+        except DatabaseError:
+            # sqlite queue is probably out of space
+            return response.raw(b"", HTTP_STATUS.INSUFFICIENT_STORAGE)
+    except PublishError:
+        # api permanently rejected this message without giving a reason
+        return response.raw(b"", HTTP_STATUS.BAD_REQUEST)
+    return response.raw(b"")
 
 
-def init_app(app: Flask):
-    """Initialize Flask app with url rules."""
+def init_app(app: Sanic):
+    """Initialize Sanic app with url rules."""
     # Use a SQLiteAckQueue because:
     # * we use acks to ensure messages only removed on success
     # * persist-queue's SQLite*Queue is faster than its Queue
     # * SQLite provides thread-safe and process-safe access
-    q = SQLiteAckQueue(**app.config.get_namespace("QUEUE_"))
+    q = SQLiteAckQueue(**{
+        key[6:].lower(): value
+        for key, value in app.config.items()
+        if key.startswith("QUEUE_")
+    })
```

Comment:
This is maybe a bit too terse. Readability would be improved if we name the dict we're defining inline. Perhaps:

```
queue_config = {key[6:]...}
q = SQLiteAckQueue(**queue_config)
```